tests/snpe_train_test.py

```diff
@@ -0,0 +1,73 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from pathlib import Path
+import tempfile
+
+import pytest
+import torch
+# from scipy.stats import beta, multivariate_normal, uniform
+# from torch import Tensor, eye, nn, ones, zeros
+from torch.distributions import Distribution, MultivariateNormal
+from torch.utils.tensorboard import SummaryWriter
+
+from sbi.inference import SNPE,  simulate_for_sbi
+from sbi.utils.user_input_checks import prepare_for_sbi
+from sbi.simulators.linear_gaussian import diagonal_linear_gaussian
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize("target_device", ("cpu", "cuda:0"))
+def test_train_with_different_data_and_training_device(target_device):
```

Comment:
It's great that you wrote this test to check your implementation, but I do not think this test is needed in the future.  The different devices are already checked [here](https://github.com/mackelab/sbi/blob/c9174e130cc622ae2219433dda541b74b37912a3/tests/inference_on_device_test.py#L30) and writing a full test whether the SummaryWriter is working seems excessive. I would rather delete it to keep our test suite clean.

Reply:
To be fair, I had (high) hopes I could check the summary writer after the fact if any data actually landed on disk. After some research on the web I cam to the conclusion, that following through with this, is more work than expected. :weary: I am happy to remove the test - even though I wouldn't agree to the statement that it pollutes the test suite. ;-) See 52d2f8d